crates/solver/src/solver/single_order_solver.rs

```diff
@@ -486,9 +484,13 @@ pub fn execution_respects_order(
     executed_sell_amount: U256,
     executed_buy_amount: U256,
 ) -> bool {
-    // note: This would be different for partially fillable orders but LimitOrder
-    // does currently not contain the remaining fill amount.
-    executed_sell_amount <= order.sell_amount && executed_buy_amount >= order.buy_amount
+    executed_sell_amount <= order.sell_amount
```

Comment:
This is incorrect for fok orders, no?
I believe the correct thing would be to first check that the executed amount makes sense (depending on partial fill and the order side) and then ensure that `provided_price >= limit_price`.

Reply:
Even if an order is marked as FOK, it has no downside to the user to sell less. This is the reasoning with which the function was originally written this way and which this PR doesn't change. If you say FOK sell 10 X for 20 Y then what you care about is getting at least 20 Y in a single trade. You don't care about selling 10 X. So if for some unexpected reason the SingleOrderSolver facilitates the trade with less sell amount then we should take it.

Reply:
Although I see now that this might not be right because it means the contract keeps the extra sell amount. (does it?). It would be more correct to sell the full sell amount and return even more buy amount.

Reply:
We should discuss this in an issue because if this is wrong then it was already wrong. This PR doesn't change it.

Reply:
> Although I see now that this might not be right because it means the contract keeps the extra sell amount. (does it?). It would be more correct to sell the full sell amount and return even more buy amount.

Yes. This was what I was thinking as well.

> Still it is unclear whether that isn't better than not trading at all if this is what the single order solver returned. It does "respect the order".

I think if we could make the even better trade work that would be nice. I just wasn't sure if this would somehow cause problems in the settlement contract because the amounts don't match.

I guess the nice thing would be to allow this better trade as legal while still forwarding the extra money to the user. But I think that could be done in a separate PR.